### ISSUE TITLE
Old version of faraday-http-cache causes gemfile conflicts

### ISSUE THREAD

**Issue by sgringwe** [launchdarkly/ruby-client#61]
Currently this gem uses 0.4 of faraday-http-cache. This conflicts with 'metainspector' which uses ~> 1.2. Is the old version required? How can I work around this to install launchdarkly?

**Comment by jkodumal** [comment]
It should be possible for us to bump our dependency to a newer version of `faraday-http-cache`-- I'll test out the change.

**Comment by sgringwe** [comment]
@jkodumal is this something you were able to test out?

**Comment by jkodumal** [comment]
Sorry @sgringwe -- this slipped through the cracks. I'll take a crack at this today.

**Comment by shyvy88** [comment]
Hi @jkodumal, I'm running into the same issues at the minute. Have you had a chance to have a look?

**Comment by shyvy88** [comment]
Forked the gem, bumped faraday-http-cache to 1.3.0, ran your specs, got 2 failures in `spec/ldclient_spec.rb` along these lines:

```
[LDClient] Unexpected exception in toggle?: #<ArgumentError: #<LaunchDarkly::ThreadSafeMemoryStore:0x00000003afd068 @cache=#<ThreadSafe::Cache:0x00000003afd040 @backend={}, @default_proc=nil>> is not a valid cache store as it does not responds to 'read', 'write' or 'delete'.> #<LaunchDarkly::ThreadSafeMemoryStore:0x00000003afd068 @cache=#<ThreadSafe::Cache:0x00000003afd040 @backend={}, @default_proc=nil>> is not a valid cache store as it does not responds to 'read', 'write' or 'delete'.
```

Looks like the interface for faraday-http-cache changed since 0.4, so it's not a simple swap out.

**Comment by shyvy88** [comment]
So your `ThreadSafeMemoryStore` class doesn't implement delete which is why this is failing. I've implemented it on our fork. Will see if it works without any further changes.

**Comment by shyvy88** [comment]
Hi @jkodumal, I've created a pull request. It's not much, but makes the gem work for us. Let me know your thoughts.

**Comment by jkodumal** [comment]
Released in version 0.8.0
